### ISSUE TITLE
[Bug]: Promise.all gives a generic error instead of surfacing the original error.

### ISSUE THREAD

**Issue by AS-Laguna** [appsmithorg/appsmith#12352]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When I use `Promise.all`, errors produced cause a generic error message: `api_call failed to execute`. Where `api_call` is the name of an API query passed to `Promise.all`.

What I expect is for the original error from `api_call` to be surfaced.

Here is the code for the buttons in the sample app:
```js
export default {
    normalError: () => {
        api_call.run(() => {}, error => {
            // 500 Internal Server Error
            showAlert(error); 
        })
    },
    promiseAllError: () => {
        const calls = [api_call.run()];
        return Promise.all(calls).catch(error => { 
            // // api_call failed to execute
            showAlert(error.message);
        })
    }
}
```

Issue, sample app, and code provided by discord user mrtillman#3732.

### Steps To Reproduce

1. Go to the public [sample app](https://app.appsmith.com/applications/6242582bc99df2369931addc/pages/6242582bc99df2369931addf).
2. Click the `Show Normal Error` button.
3. Observe `500 Internal Server Error` error.
4. Click the `Promise.all Error` button.
5. Observe `api_call failed to execute` error.

### Public Sample App

https://app.appsmith.com/applications/6242582bc99df2369931addc/pages/6242582bc99df2369931addf

### Version

Cloud v1.6.17

**Comment by stillman0000** [comment]
hey @AS-Laguna - thanks for taking the lead here. I should also mention that I added a new method to demonstrate that `Promise.allSettled` also exhibits the same behavior: 

```js
promiseAllSettledError: () => {
  const calls = [api_call.run()];
  return Promise.allSettled(calls).then((responses) => {
    for (let response of responses) {
      if (response.status == "rejected") {
        // api_call failed to execute
        showAlert(response.reason.message);
      }
    }
  });
};
```

**Comment by stillman0000** [comment]
@AS-Laguna - note: accidentally deleted the [sample app](https://app.appsmith.com/applications/6242582bc99df2369931addc/pages/6242582bc99df2369931addf). 

[here](https://appsmith-promise-error.surge.sh/promise-error-message.json) I've uploaded the app export from the [discord thread](https://discord.com/channels/725602949748752515/760761686549463060/957746977083387914).

the new sample app is https://app.appsmith.com/applications/624e407f51a8863d6c40534e/pages/624e407f51a8863d6c405350

**Comment by Nikhil-Nandagopal** [comment]
@stillman0000 that app doesn't seem to be available. Have you made it public?

**Comment by Nikhil-Nandagopal** [comment]
@stillman0000 the link you shared seems to be the issue link

**Comment by stillman0000** [comment]
> @stillman0000 the link you shared seems to be the issue link

yes, sorry I was trying to reference my previous comment where I mentioned moving the new sample app here: https://app.appsmith.com/applications/624e407f51a8863d6c40534e/pages/624e407f51a8863d6c405350

@Nikhil-Nandagopal

**Comment by Nikhil-Nandagopal** [comment]
@stillman0000 this link is not a public app. Can you please mark it as public?

**Comment by stillman0000** [comment]
hey @Nikhil-Nandagopal -  I've exported the app definition [here](https://promise-error-message.surge.sh/promise-error-message.json)

sorry, not sure why my app isn't showing as public for you:

<img width="627" alt="Screen Shot 2022-04-13 at 5 39 26 PM" src="https://user-images.githubusercontent.com/92891232/163275334-e702cbb3-6b38-47a0-855e-265a51e3d397.png">

**Comment by Nikhil-Nandagopal** [comment]
@stillman0000 thank you! This is super clear how to reproduce now
